packages/engine/Source/DataSources/ModelVisualizer.js

```diff
@@ -445,106 +442,36 @@ ModelVisualizer.prototype.getBoundingSphere = function (entity, result) {
 
   const scene = this._scene;
   const globe = scene.globe;
+  const ellipsoid = defaultValue(globe?.ellipsoid, Ellipsoid.WGS84);
 
-  // cannot access a terrain provider if there is no globe; formally set to undefined
-  const terrainProvider = defined(globe) ? globe.terrainProvider : undefined;
   const hasHeightReference = model.heightReference !== HeightReference.NONE;
-  if (defined(globe) && hasHeightReference) {
-    const ellipsoid = globe.ellipsoid;
+  if (hasHeightReference) {
     const modelMatrix = model.modelMatrix;
     scratchPosition.x = modelMatrix[12];
     scratchPosition.y = modelMatrix[13];
     scratchPosition.z = modelMatrix[14];
-    const cartoPosition = ellipsoid.cartesianToCartographic(scratchPosition);
-
-    // For a terrain provider that does not have availability, like the EllipsoidTerrainProvider,
-    // we can directly assign the bounding sphere's center from model matrix's translation.
-    if (!defined(terrainProvider.availability)) {
-      // Regardless of what the original model's position is set to, for CLAMP_TO_GROUND, we reset it to 0
-      // when computing the position to zoom/fly to.
-      if (model.heightReference === HeightReference.CLAMP_TO_GROUND) {
-        cartoPosition.height = 0;
-      }
-
-      const scratchPosition = ellipsoid.cartographicToCartesian(cartoPosition);
-      BoundingSphere.clone(model.boundingSphere, result);
-      result.center = scratchPosition;
-
-      return BoundingSphereState.DONE;
```

Comment:
Is this early return missing in the new code?

Reply:
No, the new approach does not have the same restriction.